src/expression.rs

```diff
@@ -362,26 +363,21 @@ impl<'a> FromStr for Expression {
 /// - When imaginary is set but real is 0, show only imaginary
 /// - When imaginary is 0, show real only
 /// - When both are non-zero, show with the correct operator in between
-macro_rules! format_complex {
-    ($value:expr) => {{
-        let mut operator = String::new();
-        let mut imaginary_component = String::new();
-
-        if $value.im > 0f64 {
-            operator = "+".to_owned();
-            imaginary_component = format!("{:.}i", $value.im)
-        } else if $value.im < 0f64 {
-            imaginary_component = format!("-{:.}i", $value.im)
-        }
-
-        if imaginary_component == "" {
-            format!("{:.}", $value.re)
-        } else if $value.re == 0f64 {
-            format!("{}", imaginary_component)
-        } else {
-            format!("{:.}{}{}", $value.re, operator, imaginary_component)
-        }
-    }};
+#[inline(always)]
+fn format_complex(value: &Complex64) -> String {
+    if value.re == 0f64 && value.im == 0f64 {
+        "0".to_owned()
+    } else if value.im == 0f64 {
+        ryu::Buffer::new().format_finite(value.re).to_owned()
+    } else if value.re == 0f64 {
+        ryu::Buffer::new().format_finite(value.im).to_owned() + "i"
+    } else {
+        let op = if value.im > 0f64 { "+" } else { "" };
+        ryu::Buffer::new().format_finite(value.re).to_owned()
+            + op
+            + ryu::Buffer::new().format(value.im)
+            + "i"
```

Comment:
I suppose it probably could, and I agree it looks weird. But, per [the docs](https://docs.rs/ryu/latest/ryu/struct.Buffer.html#method.new), 

> This is a cheap operation; you don’t need to worry about reusing buffers for efficiency.

If it's going to trip up others, we could reuse it, comment about it, or otherwise clean this part up.

Reply:
Ugh, the asymmetry between `format` and `format_finite` is bugging me though, hold on

Reply:
See 41c23f4